### ISSUE TITLE
Setting image caused warning about an unmounted component during an async callback

### ISSUE THREAD

**Issue by ryan2clw** [avishayil/react-native-user-avatar#54]
I ran into this with another component that I built called NetworkImage.  Basically the async call to set the image comes back after the component is unmounted, causing a yellowbox warning.  My solution was to add useRef, and just pass a boolean in as an initial value.  Forked this repo and can verify that this works.
```import React, {useState, useEffect, useRef} from 'react';
// add reference to DOM like object
  const isMounted = useRef(true);

  useEffect(() => {
    if (component) setInner(<CustomAvatar size={size} component={component} />);
    if (src) {
      const controller = new (AbortController || window.AbortController)();
      fetchImage(src, {signal: controller.signal}).then((isImage) => {
      // make sure it's still there
        if (isImage && isMounted.current) {
          setInner(
              <ImageAvatar src={src} size={size} imageStyle={imageStyle} />,
          );
        }
      });
      return () => {
        controller.abort();
        // cleanup code, prevents changing the state on an unmounted component
        isMounted.current = false;
      }
    }
  }, []);
```

**Comment by OmkarRajam** [comment]
Hi @avishayil,
I am also getting the same issue as above:-
```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in %s.%s, a useEffect cleanup function,
    in UserAvatar (at AuthorAndAnsExplTrigger.tsx:34) 
```

**Comment by avishayil** [comment]
Should be fixed with [v1.0.8](https://github.com/avishayil/react-native-user-avatar/releases/tag/v1.0.8)
